src/Analyzers/Core/CodeFixes/UseNullPropagation/AbstractUseNullPropagationCodeFixProvider.cs

```diff
@@ -168,66 +168,75 @@ private void FixIfStatement(
 
         var whenTrueStatement = (TStatementSyntax)root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan);
         var match = (TExpressionSyntax)root.FindNode(diagnostic.AdditionalLocations[2].SourceSpan, getInnermostNodeForTie: true);
+        var nullAssignmentOpt = diagnostic.AdditionalLocations.Count == 4
+            ? (TStatementSyntax?)root.FindNode(diagnostic.AdditionalLocations[3].SourceSpan, getInnermostNodeForTie: true)
+            : null;
 
         var whenPartIsNullable = diagnostic.Properties.ContainsKey(UseNullPropagationHelpers.WhenPartIsNullable);
 
         SyntaxNode nodeToBeReplaced = ifStatement;
-        SyntaxNode? replacementNode = null;
 
         // we have `if (x != null) x.Y();`.  Update `x.Y()` to be `x?.Y()`, then replace the entire
         // if-statement with that expression statement.
         var newWhenTrueStatement = CreateConditionalAccessExpression(
             syntaxFacts, generator, whenPartIsNullable, whenTrueStatement, match);
         Contract.ThrowIfNull(newWhenTrueStatement);
 
-        var isElseIf = syntaxFacts.IsElseClause(ifStatement.Parent);
```

Comment:
i reordered thids pretty heavily.  it had this local variable to indicate if we are in the `else if (x != null) x.Foo()` case.  And then lots of contorted logic doing work, but throwing in random checks of this local value.  So i just split things into an if/else where we keep all the logic for this case in one place, and then handle the normal `if` case below that.